ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/BatchingLedgerWriter.scala

```diff
@@ -45,15 +44,15 @@ class BatchingLedgerWriter(val queue: BatchingQueue, val writer: LedgerWriter)(
     */
   override def commit(
       correlationId: String,
-      envelope: kvutils.Bytes,
+      envelope: Raw.Value,
```

Comment:
Would it make sense to have a separate type `Envelope` (or even `SubmissionEnvelope`) for enveloped bytes? This would fully separate the 3 different types of `ByteString`s that we're passing around on our public APIs (i.e., keys, values, submission envelopes).

Reply:
Yes, but I believe all values are envelopes, so I think I'd rather just rename this to `Raw.Envelope`. If there is a use case for `Raw.Value` that's not an envelope, it should be highlighted at this point.

I'd also like to split `Raw.Key` in two (as documented in the PR description).

I'd like to save this for a separate change though, as this is already pretty big and I don't have time to work on it further.

Reply:
You're right -- I realized after submitting the review that all state values are wrapped into an `Envelope` anyway.